log/formatter/general_test.go

```diff
@@ -0,0 +1,294 @@
+package formatter
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	configmock "github.com/goravel/framework/contracts/config/mocks"
+)
+
+type GeneralTestSuite struct {
+	suite.Suite
+	mockConfig *configmock.Config
+	entry      *logrus.Entry
+}
+
+func TestGeneralTestSuite(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping tests of using docker")
+	}
+
+	suite.Run(t, new(GeneralTestSuite))
+}
+
+func (s *GeneralTestSuite) SetupTest() {
+	s.mockConfig = &configmock.Config{}
+	s.entry = &logrus.Entry{
+		Level:   logrus.InfoLevel,
+		Message: "Test Message",
+	}
+}
+
+func (s *GeneralTestSuite) TestFormat() {
+	s.mockConfig.On("GetString", "app.timezone").Return("UTC")
+	s.mockConfig.On("GetString", "app.env").Return("test")
+
+	general := NewGeneral(s.mockConfig)
+	tests := []struct {
+		name   string
+		setup  func()
+		assert func()
+	}{
+		{
+			name: "Error in Marshaling",
+			setup: func() {
+				s.entry.Data = logrus.Fields{
+					"root": make(chan int),
+				}
+			},
+			assert: func() {
+				formatLog, err := general.Format(s.entry)
+				s.NotNil(err)
+				s.Nil(formatLog)
+			},
+		},
+		{
+			name: "Data is not empty",
+			setup: func() {
+				s.entry.Data = logrus.Fields{
+					"root": map[string]interface{}{
+						"code":   "200",
+						"domain": "example.com",
+						"owner":  "owner",
+						"user":   "user1",
+					},
+				}
+			},
+			assert: func() {
+				formatLog, err := general.Format(s.entry)
+				s.Nil(err)
+				s.Contains(string(formatLog), "code: \"200\"")
+				s.Contains(string(formatLog), "domain: \"example.com\"")
+				s.Contains(string(formatLog), "owner: \"owner\"")
+				s.Contains(string(formatLog), "user: \"user1\"")
+			},
+		},
+	}
+
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			test.setup()
+			test.assert()
+		})
+	}
+
+	s.mockConfig.AssertExpectations(s.T())
+}
+
+func TestFormatData(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping tests of using docker")
+	}
```

Comment:
S: This judgment is only used in the Windows system, this is unnecessary if this case can be tested in Windows. The same as below.